src/open_samus_returns_rando/specific_patches/heat_room_patches.py

```diff
@@ -14,226 +22,87 @@ def _get_new_logic_shape(editor: PatcherEditor, points: int) -> Container:
-
-    points[0]["x"] = 5500.0
-    points[0]["y"] = 5900.0
-    points[1]["x"] = 17700.0
-    points[1]["y"] = 5900.0
-    points[2]["x"] = 17700.0
-    points[2]["y"] = 10700.0
-    points[3]["x"] = 5500.0
-    points[3]["y"] = 10700.0
-
-    scenario_5.raw["objects_c"]["LS_Heat_002"] = new_logic_shape
-
-    heat_actor = _get_heat_trigger(editor)
-    new_actor = editor.copy_actor(name_of_scenario, (5600.0, 16000.0, 0.0), heat_actor, "TG_Heat_033", 2)
-    new_actor["components"][0]["arguments"][16]["value"] = "LS_Heat_002"
-
-    scenario_5.add_actor_to_entity_groups("collision_camera_009", "TG_Heat_033", True)
-    scenario_5.add_actor_to_entity_groups("collision_camera_015", "TG_Heat_033", True)
-    scenario_5.add_actor_to_entity_groups("collision_camera_017", "TG_Heat_033", True)
-    scenario_5.add_actor_to_entity_groups("017_PostGamma_002", "TG_Heat_033", True)
-    scenario_5.add_actor_to_entity_groups("PostGamma_002", "TG_Heat_033", True)
-
-def _patch_area_6_b(editor: PatcherEditor):
-    name_of_scenario = "s065_area6b"
-
-    scenario_6b = editor.get_scenario(name_of_scenario)
-
-    total_boundings = ListContainer([7600.0, 5800.0, 14200.0, 7900.0])
-    new_logic_shape = _get_new_logic_shape(editor, 4)
-    new_logic_shape["data"]["total_boundings"] =  total_boundings
-    new_logic_shape["data"]["polys"][0]["boundings"] =  total_boundings
-    points = new_logic_shape["data"]["polys"][0]["points"]
-
-    points[0]["x"] = 7600.0
-    points[0]["y"] = 5800.0
-    points[1]["x"] = 14200.0
-    points[1]["y"] = 5800.0
-    points[2]["x"] = 14200.0
-    points[2]["y"] = 7900.0
-    points[3]["x"] = 7600.0
-    points[3]["y"] = 7900.0
-
-    scenario_6b.raw["objects_c"]["LS_Heat_001"] = new_logic_shape
-
-    heat_actor = _get_heat_trigger(editor)
-    editor.copy_actor(name_of_scenario, (6000.0, 9000.0, 0.0), heat_actor, "TG_Heat_003", 2)
-
-    scenario_6b.add_actor_to_entity_groups("collision_camera_014", "TG_Heat_003", True)
-    scenario_6b.add_actor_to_entity_groups("collision_camera_017", "TG_Heat_003", True)
-    scenario_6b.add_actor_to_entity_groups("collision_camera_018", "TG_Heat_003", True)
-    scenario_6b.add_actor_to_entity_groups("PostGamma_002", "TG_Heat_003", True)
-
-
-def _patch_area_6_c(editor: PatcherEditor):
-    name_of_scenario = "s067_area6c"
-
-    scenario_6c = editor.get_scenario(name_of_scenario)
-
-    total_boundings = ListContainer([-27100.0, 3400.0, -21900.0, 6000.0])
-    new_logic_shape = _get_new_logic_shape(editor, 4)
-    new_logic_shape["data"]["total_boundings"] =  total_boundings
-    new_logic_shape["data"]["polys"][0]["boundings"] =  total_boundings
-    points = new_logic_shape["data"]["polys"][0]["points"]
+    points[0]["x"] = point[0][0]
+    points[0]["y"] = point[0][1]
+    points[1]["x"] = point[1][0]
+    points[1]["y"] = point[1][1]
+    points[2]["x"] = point[2][0]
+    points[2]["y"] = point[2][1]
+    points[3]["x"] = point[3][0]
+    points[3]["y"] = point[3][1]
```

Comment:
`points` and `point` is confusing because both is a list of points.
At that point (pun intended 😛), I would just move it to `_get_new_logic_shape`, pass the list of points instead of the amount and create them directly with the correct coordinates.

Reply:
Move the entire boundings and points section? Don't I have to reference the class still?

Reply:
Yes, either that or pass the list of points like I mentioned. Which made me realise that the type is wrong.

If you want to pass the whole reference like you mentioned. Something like that (untested):
```python
def _get_new_logic_shape(editor: PatcherEditor, new_actor: NewHeatActor) -> Container:
    scenario_2b = editor.get_scenario("s025_area2b")
    ls_copy = copy.deepcopy(scenario_2b.raw["objects_c"]["LS_Heat_001"])
    example_point = ls_copy["data"]["polys"][0]["points"][0]
    ls_copy["data"]["polys"][0]["points"].clear()
    for point in new_actor.points:
        new_point = copy.deepcopy(example_point)
        new_point["x"] = point[0]
        new_point["y"] = point[1]
        ls_copy["data"]["polys"][0]["points"].append(new_point)
    ls_copy["data"]["polys"][0]["num_points"] = len(new_actor.points)
    return ls_copy
```